sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -110,19 +110,25 @@ case class CreateViewCommand(
 
   private def isTemporary = viewType == LocalTempView || viewType == GlobalTempView
 
-  // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS' to be consistent with 'CREATE TEMPORARY TABLE'
-  if (allowExisting && isTemporary) {
-    throw new AnalysisException(
-      "It is not allowed to define a TEMPORARY view with IF NOT EXISTS.")
-  }
+  if (isTemporary) verifyTempView()
+
+  private def verifyTempView(): Unit = {
+    // Disallows 'CREATE TEMPORARY VIEW IF NOT EXISTS'
+    // to be consistent with 'CREATE TEMPORARY TABLE'
+    if (allowExisting) {
```

Comment:
Does this need to be predicated on whether the postgresql dialect is used?

Reply:
Postgres doesn't allow using 'IF NOT EXISTS' while creating view. 
https://www.postgresql.org/docs/9.2/sql-createview.html

Reply:
So this is the same whether or not postgresql dialect is used?

Reply:
Yes